credentials/local/local_test.go

```diff
@@ -118,56 +118,41 @@ type testServerHandleResult struct {
 }
 
 // Server accepts a client's connection with local credentials.
-func serverHandle(t *testing.T, hs serverHandshake, done chan testServerHandleResult, lis net.Listener) {
+func serverHandle(hs serverHandshake, done chan testServerHandleResult, lis net.Listener) {
 	serverRawConn, err := lis.Accept()
 	serverAuthInfo, err := hs(serverRawConn)
 	if err != nil {
-		t.Errorf("Server failed while handshake. Error: %v", err)
 		serverRawConn.Close()
-		done <- testServerHandleResult{authInfo: nil, err: err}
+		done <- testServerHandleResult{authInfo: nil, err: fmt.Errorf("Server failed while handshake. Error: %v", err)}
 	}
 	done <- testServerHandleResult{authInfo: serverAuthInfo, err: nil}
 }
 
-func ServerAndClientHandshake(t *testing.T, network, listAddr string) credentials.SecurityLevel {
+func serverAndClientHandshake(lis net.Listener) (credentials.SecurityLevel, error) {
 	done := make(chan testServerHandleResult, 1)
 	timeout := time.After(100 * time.Second)
```

Comment:
Is the leakchecker happy with the use of `time.After`? Because `time.After`'s doc string says the following:
The underlying Timer is not recovered by the garbage collector until the timer fires. If efficiency is a concern, use NewTimer instead and call Timer.Stop if the timer is no longer needed.